### PR TITLE
Handle InMemory files in recent file list

### PR DESCRIPTION
**Pull request by swmitra**
This PR adds the capability of handling InMemoryFile as part of the navigation list. When the file gets cleared from Workingset , it gets cleared from Navigation list as well. 

Ping @nethip for review.

### CODE REVIEW

**Comment by nethip** [comment]
@swmitra The changes look good, but there are some indentation issues. Please indent the newly added code so that I can go ahead with the merge.

**Comment by swmitra** [comment]
@nethip Thanks a lot for the quick review :+1: 
Fixed the tab spacing issues.

File: src/extensions/default/NavigationAndHistory/main.js [hunk 1]
```diff
@@ -164,14 +164,25 @@ define(function (require, exports, module) {
...
 
-        fileEntry.exists(function (err, exists) {
-            if (!err && exists) {
-                deferred.resolve();
-            } else {
-                _mrofList[index] = null;
-                deferred.reject();
-            }
-        });
+        if (entry.inMem) {
```
Comment (nethip) [inline src/extensions/default/NavigationAndHistory/main.js]: I think this piece still needs indentation.

**Comment by nethip** [comment]
Thanks @swmitra .

**Pull request merged**